src/frontend/src/expr/mod.rs

```diff
@@ -772,7 +772,14 @@ impl ExprImpl {
                     ExprType::Add | ExprType::Subtract => {
                         let (_, lhs, rhs) = function_call.clone().decompose_as_binary();
                         if let ExprImpl::InputRef(input_ref) = &lhs && rhs.is_const() {
-                            Some((input_ref.index(), Some((expr_type, rhs))))
+                            if rhs.return_type() == DataType::Interval && rhs.as_literal().map_or(true, |literal| literal.get_data().as_ref().map_or(false, |scalar| {
```

Comment:
Checking my understanding:
* The default `true` after `as_literal` means we return `None` for non-literal
* The default `false` after `get_data` means we return `Some` when it is literal `null`

Maybe add some comments or tests to help make this clear.

BTW:
* The closed PR #9647 updates `is_now_offset`, and here we update `as_input_offset`. How are they related? Do we need to fix both?
* `is_const` then `as_literal` misses the case when rhs is `interval + interval` or `interval * number`. Practically it should be rare. Just making the difference clear.

Reply:
> Checking my understanding:
> 
> * The default `true` after `as_literal` means we return `None` for non-literal
> * The default `false` after `get_data` means we return `Some` when it is literal `null`
> 
> Maybe add some comments or tests to help make this clear.
> 
> BTW:
> 
> * The closed PR [feat(now): streaming now() with a timezone #9647](https://github.com/risingwavelabs/risingwave/pull/9647) updates `is_now_offset`, and here we update `as_input_offset`. How are they related? Do we need to fix both?
> * is_const then as_literal misses the case when rhs is interval + interval or interval * number. Practically it should be rare. Just making the difference clear.
Yes, it's the same as your understanding.
I have just added some explanation, thanks.

> How are they related? Do we need to fix both?
IIUC, even if we do not have monotonicity, we can still handle `now`, although we cannot do state cleaning in that case. Therefore we do not need to fix `is_now_offset`.
On the other hand, `as_input_offset` involves in some calculation on watermark so we need to fix this.

> is_const then as_literal misses the case when rhs is interval + interval or interval * number. Practically it should be rare. Just making the difference clear.

I've added an example.